agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -464,5 +445,59 @@ public class OpenTelemetry {
 
             //endregion Tool Call
         }
+
+        private fun setMessageAttributes(span: Span, index: Int, message: Message, config: OpenTelemetryConfig, type: String) {
```

Comment:
major: @agbragin, could you please clarify if it is expected to set the same set of attributes in both "before..." and "after..." handlers? Why do we need to duplicate this logic here?

Reply:
The idea to is have both client (user, system) and server (assistant) messages traced, but now I understand that I likely had a wrong expectations regarding `eventContext.prompt.messages`. 
It looks like `interceptAfterLLMCall` `eventContext.prompt.messages` contains all the messages from `interceptBeforeLLMCall` `eventContext.prompt.messages` **plus** messages added by the assistant, right?
Then we need to substract messages from `interceptBeforeLLMCall` on `interceptAfterLLMCall`.

Reply:
>It looks like interceptAfterLLMCall eventContext.prompt.messages contains all the messages from interceptBeforeLLMCall eventContext.prompt.messages plus messages added by the assistant, right?

That's correct.